lib/tar.ml

```diff
@@ -555,12 +553,12 @@ module Header = struct
     if level = Ustar || (level = GNU && x.devmajor = 0 && x.devminor = 0) then begin
       if level = Ustar then begin
         set_hdr_magic c "ustar";
-        set_hdr_version c 0;
+        set_hdr_version c "00";
       end else begin
         (* OLD GNU MAGIC: use "ustar " as magic, and another " " in the version *)
-        let t = sizeof_hdr_magic + sizeof_hdr_version in
-        let magic = marshal_string "ustar  " t in
-        Cstruct.blit_from_string magic 0 c hdr_magic_off t
+        let magic_len = sizeof_hdr_magic + sizeof_hdr_version in
+        let magic = marshal_string "ustar  " magic_len in
+        Cstruct.blit_from_string magic 0 c hdr_magic_off magic_len
```

Comment:
now that set_hdr_version no longer requires an int, we can actually do:
```OCaml
        set_hdr_magic c "ustar ";
        set_hdr_version c " \000";
```

-> set the magic to `ustar ` and the version to ` \000`. WDYT?